doc/user/content/ops/speedup.md

```diff
@@ -0,0 +1,149 @@
+
+```sql
+CREATE TABLE contacts (
+    first_name TEXT,
+    last_name TEXT,
+    phone INT,
+    prefix INT
+);
+
+INSERT INTO contacts SELECT 'random_name', 'random_last_name', generate_series(0, 100000), 1;
+```
+
+<!-- This will be removed till the PR makes it into production and is available to everyone -->
+<!-- ### Checking if a query is using an index
+
+The [EXPLAIN](https://materialize.com/docs/sql/explain/#conceptual-framework) command displays if the query plan is including an index to read from.
+
+```sql
+EXPLAIN SELECT * FROM contacts WHERE first_name = 'Jann';
+```
+
+```
+                      Optimized Plan
+------------------------------------------------------------------
+ %0 =                                                            +
+ | ReadExistingIndex materialize.public.contacts_first_name_idx  +
+ | Get materialize.public.contacts (u23)                         +
+ | Filter (#0 = "Jann")                                          +
+```
+
+`ReadExistingIndex` indicates that the query is using the `contacts_first_name_idx`. An absence of it would mean
+that the query is scanning the whole table resulting in slower results. -->
+
+## WHERE
+
+The filtering clause is one of the most used operations in any database. Set up an index over the columns that a query filters by to increase the speed.
+
+### Literal Values
+
+Filtering by literal values is a typical case. An index over the column filtered will do the work.
+
+Back to our example, to improve retrieving all the contacts by a particular first name (the literal value):
+
+```sql
+CREATE INDEX contacts_first_name_idx ON contacts (first_name);
+
+SELECT * FROM contacts WHERE first_name = 'Charlie';
+```
+
+### Expressions
+
+Expressions can also be part of the index. Materialize will calculate them faster using an index instead of calculating them on the fly for every query.
+
+E.g., Since contact names are probably not always correctly written, using a function to upper case the name is helpful.
+
+```sql
+CREATE INDEX contacts_upper_first_name_idx ON contacts (upper(first_name));
+
+SELECT * FROM contacts WHERE upper(first_name) = 'CHARLIE';
+```
+
+An _expression_ goes beyond a function. It could be a mathematical expression that is present in the filtering
+
+```sql
+CREATE INDEX contacts_upper_first_name_idx ON contacts (prefix - 1 = 0);
+
+SELECT * FROM contacts WHERE prefix - 1 = 0;
+```
+
+
+### Literal Values and Expressions
+
+It's possible to also combine both worlds, literal values and expressions.
+
+```sql
+CREATE INDEX contacts_upper_first_name_phone_idx ON contacts (upper(first_name), phone);
+
+SELECT * FROM contacts WHERE first_name = 'CHARLIE' AND phone = 873090;
+```
+
+### Compound index
+
+When using a compound index, consider how the optimizer works. It's rule-based, and if it detects an index it is helpful for a query, it will use it. Creating an index over all the columns doesn't mean that all
```

Comment:
This should start with defining what a compound index is before talking about how to use one. "consider how the optimizer works" assumes that the user already knows how the optimizer works, but we're trying to teach them that here. I think that phrase should be removed. This feature is called "multicolumn indexes" in postgres, and we should probably stick with that. When talking about the rule-based stuff and how it might use it if helpful, we need to tell the users the criteria there instead of make them guess. Does the `WHERE` clause need to match a prefix of the columns, any subset of the columns, at least all of the columns, or exactly all of the columns?

Reply:
@mjibson, from my usage experience, is when it matches all the columns, and I'm skeptical about the subset.

> Does the WHERE clause need to match a prefix of the columns, any subset of the columns, at least all of the columns, or exactly all of the columns?

@aalexandrov, do you know precisely when the optimizer/planner decides to use an index?

Reply:
@mjibson I've added the `AND` condition. The `IN` operator seems to also gets the boost. I felt that the operator part should be described on the `Create Index` documentation page rather than in the optimizations.

Reply:
Hey @wangandi, may I ask you for a comment/guidance here?

Reply:
The `IN` operator will get the boost currently if there is only one item in the `IN`. @ggevay is working right now on the boost when there are multiple items in the `IN`. 

For `=` `AND`ed together, you need at least all of the columns. So if your index is (substr(col1, 3), col2), you need to have `substr(col1, 3) = 'zzz' AND col2 = 4` to make use of the index, but if you have an extra `AND col3 = 5`, you can still use the index; the extra `col3 =5` will just run after the index sped up the first two filters.

Reply:
I'm almost finished with enabling index usage for an OR of literal constraints, e.g.
`(f1 = 3 AND f2 = 5) OR (f1 = 7 AND f2 = 9)`. This will also cover an IN list with multiple items, e.g.
`(f1,f2) IN ((3,5), (7,9))`.

Reply:
@joacoc, the [IN list epic](https://github.com/MaterializeInc/materialize/issues/13151) is done. How should we document it?
- I push a commit to this PR, or
- you change this PR, or
- I wait until you merge this PR, and then I document the changes in a follow-up PR?

Reply:
@ggevay Super! I would opt for 3) after this branch merges with the platform docs branch.

This PR is over the `lts-docs` branch, and the docs in this branch relate to the binary (I'm assuming the new feature appears only in the "platform" version)

Reply:
@joacoc Ok, then please ping me when I should start working on the follow-up PR.